src/compiler/interface.jl

```diff
@@ -73,7 +73,8 @@ julia> gradient([7, 11], 0, 1) do x, y, d
 """
 function gradient(f, args...)
   y, back = pullback(f, args...)
-  return back(sensitivity(y))
+  grad = back(sensitivity(y))
+  map(_project, args, grad)
```

Comment:
do we want this at the `gradient` or the `pullback` level?

Reply:
My thinking was to start small! Applying it to `gradient` applies it to the user-facing calls, once. Applying it to `pullback` or `_pullback` inserts it into many more places internally... maybe it'll make `sin'''(1.0)` unhappy.